src/nbl/video/COpenGL_LogicalDevice.h

```diff
@@ -352,18 +437,17 @@ class COpenGL_LogicalDevice : public IOpenGL_LogicalDevice
         m_threadHandler.template waitForRequestCompletion<SRequestInvalidateMappedMemoryRanges>(req);
     }
 
-    void* mapMemory(const IDriverMemoryAllocation::MappedMemoryRange& memory, IDriverMemoryAllocation::E_MAPPING_CPU_ACCESS_FLAG access = IDriverMemoryAllocation::EMCAF_READ_AND_WRITE) override final
+    void* mapMemory(const IDriverMemoryAllocation::MappedMemoryRange& memory) override final
```

Comment:
nope, even though buffer is from a memory type that supports RW, I still might want to map with a subset of usages.

so leave an extra parameters for narrowing down the mapping access (and bitwise AND it with the actual memory type capability so its a strict subset).

Write in the docs that that some APIs (like Vulkan ignore it) and we AND the caps anyway.